packages/block-library/src/comments/index.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Server-side rendering fallback of the `core/comments-query-loop` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/comments-query-loop` block fallback on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ * @return string Returns the filtered post comments for the current post wrapped inside "p" tags.
+ */
+function render_block_core_comments_query_loop( $attributes, $content, $block ) {
+	global $post;
+
+	// Do we have child blocks?
+	if ( count( $block->inner_blocks ) > 0 ) {
+		$inner_blocks_html = '';
+		foreach ( $block->inner_blocks as $inner_block ) {
+			$inner_blocks_html .= $inner_block->render();
+		}
+		return $inner_blocks_html;
+	}
+
+	$post_id = $block->context['postId'];
+	if ( ! isset( $post_id ) ) {
+		return '';
+	}
+
+	$comment_args = array(
+		'post_id' => $post_id,
+		'count'   => true,
```

Comment:
Don't we need to check the comment status here?

Reply:
Added by @DAreRodz in b9b69d227fcf8cd279f0aedc16c30332bc5dd520.